### ISSUE TITLE
File viewer shows un-shelved and un-deliverable files

### ISSUE THREAD

**Issue by cbeer** [sul-dlss/sul-embed#332]
Here's an example object:

http://purl.stanford.edu/mr497sx5638

The permissions document probably should not be displayed in the embed file list. I have a ticket in (LEGACY-1589) to validate the public xml is correct, but it seems like sul-embed might exclude unshelved files as well..

**Comment by blalbrit** [comment]
Hi Chris,

This particular object is EEMS-specific. The public XML is correct, but for EEMS materials depositors are allowed to set the content as non-deliverable if they are still pursuing permissions (or have received negative).

Normally, only shelved files will appear in PURL (see http://purl-test.stanford.edu/pd817cv1860.xml for example).

Agree that embed should exclude cases where unshelved files appear in PURL.xml.

**Comment by mejackreed** [comment]
So @blalbrit is the logic below correct?
```ruby
unless <file shelve="no">
  display!
end
```

**Comment by cbeer** [comment]
Or is there value in showing we have an unshelved, undeliverable file?

**Comment by mejackreed** [comment]
So we are doing something similar with embargoed items, see:

https://github.com/sul-dlss/sul-embed/pull/322

I'm not sure if saying "this file is not shelved" works, is there something more understandable about what this actually means that we could use?

**Comment by blalbrit** [comment]
nervously... "maybe"?

For EEMS, 
unless <file shelve="no"> or <file deliver="no">
  display!
end

And obeying all higher-order object rights, etc. etc. We should check with Lynn when he gets back, but that should do the trick.

B.

**Comment by mejackreed** [comment]
Ok tagging @LynnMcRae here for comment.

**Comment by blalbrit** [comment]
And sorry for the missing angle-brackets... was writing in a hurry.

**Comment by blalbrit** [comment]
```ruby
unless <file shelve="no"> or <deliver="no">
  display!
end
```

**Comment by LynnMcRae** [comment]
The “deliver” attribute was replaced by the “publish” attribute years ago. Is EEMs really still producing “deliver”?  Regardless, publish and deliver should both be factored in a test, as well as the absence of these attributes (an explicit no is not guaranteed as only an explicit yes is actionable).

Forget all thet. I just realized the problem is deeper and earlier. If a does not have publish=yes (and possibly deliver=yes for backward compatibility) it is supposed to be stripped from the public XML by publish. That is the very definition of what it means to publish a file or not. The example in the ticket should be republished to see if it changes.   I was going to try but the object had never actually been accessioned.

Lynn
